### PR TITLE
[WIP Do not merge] Add GH workflow for fornightly release on every 1&3 Friday at 09:00 hrs

### PR DESCRIPTION
**Pull request by husky-parul**

### CODE REVIEW

**Review by rootfs — APPROVED** [review]

**Comment by SamYuan1990** [comment]
.... is this PR status as WIP Don't merge or we are ready to merge it? @rootfs / @husky-parul ?

**Comment by husky-parul** [comment]
I forgot to remove the labels. Since it was approved I merged it. Did you have any comments? I can incorporate them in another PR. LMK. Thanks

File: .github/workflows/fornightly-release.yml [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+      with:
+        image: quay.io/sustainable_computing_io/kepler:${{ env.RELEASE_NAME }}
+        artifact-name: sbom-kepler- ${{ env.RELEASE_NAME }}.json
+        output-file: ./sbom-kepler- ${{ env.RELEASE_NAME }}.spdx.json
+
+    - name: Attach SBOM to release
+      uses: softprops/action-gh-release@v1
+      with:
+        tag_name: ${{ env.TAG_NAME }}
+        files: ./sbom-kepler- ${{ env.RELEASE_NAME }}.spdx.json
```
Comment (SamYuan1990) [inline .github/workflows/fornightly-release.yml:146]: where are we publish those SBOM files?

File: .github/workflows/fornightly-release.yml [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+          
+      - name: release rpm
+        uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+        with:
+          upload_url: ${{ steps.create_release.outputs.upload_url }}
+          asset_path: /tmp/kepler.rpm.tar.gz
+          asset_name: kepler.rpm.tar.gz
+          asset_content_type: application/gzip
```
Comment (SamYuan1990) [inline .github/workflows/fornightly-release.yml:104]: do we just refresh the release page or we create a new one?

**Comment by SamYuan1990** [comment]
is this pr works for rpm and SBOM? I am interested with the question.

File: .github/workflows/fornightly-release.yml (hunk 2 above)
Comment (husky-parul) [reply .github/workflows/fornightly-release.yml:104]: As of now we just put it as a part of release artifacts.

Comment (husky-parul) [reply .github/workflows/fornightly-release.yml:104]: https://github.com/sustainable-computing-io/kepler/releases/tag/v0.5.1

**Comment by husky-parul** [comment]
> is this pr works for rpm and SBOM? I am interested with the question.
@SamYuan1990 The PR will create the following release artifacts 

- kepler-rpm
- sbom file
- Source code as zip and 
- Source code as tar.gz
- pushes two images in the quay with tags `latest` and `<release-version>`

The naming convention for release version is as follows: 

```
fornightly-$(date +'%Y-%m-%d')
```

Please note we are not creating release branches for fortnightly. Only release and tags

**Comment by SamYuan1990** [comment]
> > is this pr works for rpm and SBOM? I am interested with the question.
> > @SamYuan1990 The PR will create the following release artifacts
> 
> * kepler-rpm
> * sbom file
> * Source code as zip and
> * Source code as tar.gz
> * pushes two images in the quay with tags `latest` and `<release-version>`
> 
> The naming convention for release version is as follows:
> 
> ```
> fornightly-$(date +'%Y-%m-%d')
> ```
> 
> Please note we are not creating release branches for fortnightly. Only release and tags

could you please share me a link to check those files?
https://github.com/sustainable-computing-io/kepler/releases ?

File: .github/workflows/fornightly-release.yml (hunk 2 above)
Comment (husky-parul) [reply .github/workflows/fornightly-release.yml:104]: > do we just refresh the release page or we create a new one?

As of now we  do not refresh the release we are using the naming conventions as `fornightly-$(date +'%Y-%m-%d')`
But refreshing a release seems like a better idea to reduces the artifacts we push on github. I am open to refreshing the release.

**Comment by husky-parul** [comment]
> > > is this pr works for rpm and SBOM? I am interested with the question.
> > > @SamYuan1990 The PR will create the following release artifacts
> > 
> > 
> > 
> > * kepler-rpm
> > * sbom file
> > * Source code as zip and
> > * Source code as tar.gz
> > * pushes two images in the quay with tags `latest` and `<release-version>`
> > 
> > The naming convention for release version is as follows:
> > ```
> > fornightly-$(date +'%Y-%m-%d')
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Please note we are not creating release branches for fortnightly. Only release and tags
> 
> could you please share me a link to check those files? https://github.com/sustainable-computing-io/kepler/releases ?

This workflow was never run so no resources have been created by the workflow yet but it would be similar to this 
https://github.com/sustainable-computing-io/kepler/releases/tag/v0.5.1

File: .github/workflows/fornightly-release.yml (hunk 1 above)
Comment (husky-parul) [reply .github/workflows/fornightly-release.yml:146]: As of now we just put it as a part of release artifacts.

https://github.com/sustainable-computing-io/kepler/releases/tag/v0.5.1

**Comment by SamYuan1990** [comment]
> > > > is this pr works for rpm and SBOM? I am interested with the question.
> > > > @SamYuan1990 The PR will create the following release artifacts
> > > 
> > > 
> > > 
> > > * kepler-rpm
> > > * sbom file
> > > * Source code as zip and
> > > * Source code as tar.gz
> > > * pushes two images in the quay with tags `latest` and `<release-version>`
> > > 
> > > The naming convention for release version is as follows:
> > > ```
> > > fornightly-$(date +'%Y-%m-%d')
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > Please note we are not creating release branches for fortnightly. Only release and tags
> > 
> > 
> > could you please share me a link to check those files? https://github.com/sustainable-computing-io/kepler/releases ?
> 
> This workflow was never run so no resources have been created by the workflow yet but it would be similar to this https://github.com/sustainable-computing-io/kepler/releases/tag/v0.5.1

yea... that's where I wanna to confirm and figure out.
1. for container image, as we can re tag latest and push, hence this PR as fornightly works.
2. for release page. we could not over write  https://github.com/sustainable-computing-io/kepler/releases/tag/v0.5.1
further I assume for the 1st time running, it will create a new release page with name as latest?
for the 2nd time and next running, it will re create the page or just update it?

**Comment by SamYuan1990** [comment]
@husky-parul , or if the release page is just timestamp, then I would like to hold this PR by revert. as I am not sure what's our version/release management? cc: @rootfs 
I suppose that related one or many of OpenSSF questions as release management.

**Comment by husky-parul** [comment]
@rootfs to be clear using `pre-release` for `fornightly` implies that we will keep refreshing the `pre-release` page on github and quay image for fornightly build? Also does it mean that fornightly will not create `latest` image tag for quay.io ?

**Comment by rootfs** [comment]
@husky-parul yes. Let's have more discussions in the meeting tomorrow.

**Pull request merged**